contentcuration/search/viewsets/contentnode.py

```diff
@@ -151,71 +152,43 @@ class SearchContentNodeViewSet(ContentNodeViewSet):
+        queryset = super(SearchContentNodeViewSet, self).get_queryset()
+        return ContentNode._annotate_channel_id(queryset)
+
     def annotate_queryset(self, queryset):
         """
         1. Do a distinct by 'content_id,' using the original node if possible
         2. Annotate lists of content node and channel pks
         """
-        # Get accessible content nodes that match the content id
-        content_id_query = ContentNode.filter_view_queryset(ContentNode.objects.all(), self.request.user).filter(
-            content_id=OuterRef("content_id")
-        )
-
-        # Combine by unique content id
-        deduped_content_query = (
-            content_id_query.filter(content_id=OuterRef("content_id"))
-            .annotate(
-                is_original=Case(
-                    When(original_source_node_id=F("node_id"), then=Value(1)),
-                    default=Value(2),
-                    output_field=IntegerField(),
-                ),
-            )
-            .order_by("is_original", "created")
-        )
-        queryset = queryset.filter(
-            pk__in=Subquery(deduped_content_query.values_list("id", flat=True)[:1])
-        ).order_by()
-
         thumbnails = File.objects.filter(
             contentnode=OuterRef("id"), preset__thumbnail=True
         )
 
-        descendant_resources = (
-            ContentNode.objects.filter(
-                tree_id=OuterRef("tree_id"),
-                lft__gt=OuterRef("lft"),
-                rght__lt=OuterRef("rght"),
-            )
-            .exclude(kind_id=content_kinds.TOPIC)
-            .values("id", "role_visibility", "changed")
-            .order_by()
-        )
-        original_channel_name = Coalesce(
-            Subquery(
-                Channel.objects.filter(pk=OuterRef("original_channel_id")).values(
-                    "name"
-                )[:1]
-            ),
-            Subquery(
-                Channel.objects.filter(main_tree__tree_id=OuterRef("tree_id")).values(
-                    "name"
-                )[:1]
-            ),
+        descendant_resources_count = ExpressionWrapper(((F("rght") - F("lft") - Value(1)) / Value(2)), output_field=IntegerField())
+
+        channel_name = Subquery(
+            Channel.objects.filter(pk=OuterRef("channel_id")).values(
+                "name"
+            )[:1]
         )
+
         queryset = queryset.annotate(
-            location_ids=SQArrayAgg(content_id_query, field="id"),
-            resource_count=SQCount(descendant_resources, field="id"),
+            resource_count=descendant_resources_count,
             thumbnail_checksum=Subquery(thumbnails.values("checksum")[:1]),
             thumbnail_extension=Subquery(
                 thumbnails.values("file_format__extension")[:1]
             ),
             content_tags=NotNullMapArrayAgg("tags__tag_name"),
-            original_channel_name=original_channel_name,
+            original_channel_name=channel_name,
```

Comment:
This seems like a change that introduces a regression. The query producing this annotation has changed

Reply:
This indeed introduces a regression. This regression is intended. I have removed the logic for producing unique content nodes. Copied nodes (nodes with same content id but on different channels) will also be present in the result set. So channel name should be the name of the channel the content node is in right now.

We do not need to tell the user about the name of channel from where it was copied. We just need to let them know about the channel it is in right now.

Does this make sense @bjester? What do you think?

Reply:
This changes the meaning of `original_channel_id` - I think if we need to remove this field for performance reasons, then we should remove it, rather than update it to something inaccurate. We should check with @jtamiace to see how best to handle this in the user interface if this is a strong performance concern.

Reply:
Right now, we should keep this on hold until we evaluate our performance of new full text search. I am hoping that our search will be efficient enough to allow us to run de-duplication of content_ids and then we will be able to keep our original_channel_name field intact.

Reply:
@vkWeb To clarify, what do you mean by "we should keep this on hold"?

Reply:
@bjester By on hold, I meant that the decision of removing the existing logic of `original_channel_name` should be taken after I fully implement full text search and test it on develop database.

Reply:
I want to us to take informed decisions based on our real world performance benchmarks.

Reply:
I have updated the queryset to annotate the original channel name. 

But I have not de-duplicated the query because two contentnodes with the same `content_id` can have entirely different content inside, we don't know what node the user wants to import so probably we should not de-duplicate. For e.g. an exercise was imported from a published channel and we completely changed the questions inside it but kept the metadata as is, upon search both nodes will show up, how do we decide which node to show and which to discard since both of them have completely different content...?

Instead in a future PR we should rank the results appropriately and display the top most 5000 results maybe? That would be more helpful from user's perspective in my opinion sir.

What are your thoughts sir?

Reply:
> But I have not de-duplicated the query because two contentnodes with the same content_id can have entirely different content inside, we don't know what node the user wants to import so probably we should not de-duplicate

The fact that this is true is a long standing bug in Studio, most acutely for exercises, but also applicable to other resources: https://github.com/learningequality/studio/issues/1055

Reply:
woah, I thought its a desired behaviour... :eye: btw on what scenarios we care if two nodes have exact same content inside? :thinking:

Reply:
The `content_id` is used to track progress on a specific resource in Kolibri, so it has quite an important role, and when the value is conflated across multiple resources it causes issues.

Reply:
Oh, that's pretty important, then we should fix that bug I think, if the node gets modified we should change the `content_id`.